NuRadioReco/modules/voltageToEfieldConverter.py

```diff
@@ -137,14 +138,15 @@ def run(self, evt, station, det, debug=False, debug_plotpath=None, use_channels=
         event_time = station.get_station_time()
         station_id = station.get_id()
 
-        if use_MC_direction:
-            zenith = station.get_sim_station()[stnp.zenith]
-            azimuth = station.get_sim_station()[stnp.azimuth]
+        if use_MC_direction and evt.has_sim_shower():
+            sim_shower = list(evt.get_sim_showers())[0]
+            zenith = sim_shower[shp.zenith]
+            azimuth = sim_shower[shp.azimuth]
             sim_present = True
         else:
-            logger.info("Using reconstructed (or starting) angles as no signal arrival angles are present")
-            zenith = station[stnp.zenith]
-            azimuth = station[stnp.azimuth]
+            shower = evt.get_first_shower()
```

Comment:
this also seems confusing. It seems that you can create several sim showers associated with the station. This will lead to not well defined efield reconstructions.